File: client/src/entities/PostComment.ts
import PageResponse from "./PageResponse";
import { UserData } from "./User";

export interface PostComment extends UserData {
  postCommentId: number;
  comment: string;
  commentImage: string;
  timestamp: string;
}

export default interface PostCommentListResponse {
  postCommentList: PostComment[];
  pageResponse: PageResponse;
}

File: server/src/main/java/com/facebook/server/entity/PostComment.java
package com.facebook.server.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.awt.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "post_comment")
public class PostComment extends Timestamp {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long postCommentId;

    private String comment;
    private String commentImage;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "post_id")
    private Post post;

    // @ManyToOne
    // @JoinColumn(name = "image_id")
    // private PostImage postImage;
}
